File: src/main/java/racingcar/model/car/Car.java
package racingcar.model.car;

import racingcar.common.ErrorMessage;
import racingcar.model.number.Number;

public class Car {
    private static final int MAX_NAME_LENGTH = 5;
    private final String name;
    private final Position position;

    public Car(final String name) {
        validateNameLength(name);
        this.name = name;
        this.position = new Position();
    }

    private void validateNameLength(final String name) {
        if (name.length() > MAX_NAME_LENGTH) {
            throw new IllegalArgumentException(ErrorMessage.NAME_LENGTH_ERROR.getMessage());
        }
    }

    public void goOrStop(final Number number) {
        if (number.isGreaterThanForwardThreshold()) {
            position.increasePosition();
        }
    }

    public Position getPosition() {
        return position;
    }

    public String getName() {
        return name;
    }
}

File: src/main/java/racingcar/model/car/Car.kt
package racingcar.model.car

import racingcar.model.number.Number
import java.util.*

class Car(name: String) {
    private val name = Name(name)
    val position: Position = Position()

    fun goOrStop(number: Number) {
        if (number.isGreaterThanForwardThreshold) {
            position.increase()
        }
    }

    fun getName(): String {
        return name.name
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || javaClass != o.javaClass) {
            return false
        }
        val car = o as Car
        return name == car.name
    }

    override fun hashCode(): Int {
        return Objects.hashCode(name)
    }
}
